exonum-java-binding/doc/Migration_guide_0.9.md

```diff
@@ -0,0 +1,183 @@
+operation and changes how proofs need to be verified.
+
+In some cases (rarer than is usually considered), it might be needed to **not** hash the keys
+(use them as _internal_ keys). Such `ProofMap` variant is still supported, see if your
+service implementation fits the [*requirements*][proof-map-non-hashing] to keep using it.
+If it doesn't — please migrate your schema to use the new default one.
+
+[proof-map]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.html
+[proof-map-non-hashing]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.html#key-hashing
+
+#### Disable Proofs
+
+Proofs are temporarily disabled. Creating proofs is _not_ supported in this release.
+They will be re-enabled in one of the next releases.
+
+### Update the Transactions
+
+1. Update the [`TransactionConverter#toTransaction`][to-transaction]
+to its new signature.
+
+2. Use `int` ids in transactions.
+
+3. When needed to access the newly namespaced schema in `Transaction#execute`,
+use [`TransactionContext.getServiceName`][tx-context-get-name]
+or `TransactionContext.getServiceId`:
+
+```java
+  @Override
+  public void execute(TransactionContext context) throws TransactionExecutionException {
+    MySchema schema = new MySchema(context.getFork(), context.getServiceName());
+    // ...
+  }
+```
+
+4. Remove `Transaction#info` implementation: this method is no longer provided.
+The core returns complete transaction messages in response to transaction request,
+which can be decoded by the client using appropriate protobuf declarations.
+
+[to-transaction]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/service/TransactionConverter.html
+[tx-context-get-name]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/transaction/TransactionContext.html#getServiceName--
+
+### Update the Integration Tests
+
+The integration tests using Exonum Testkit to create a test network with a service-under-test need
+to be updated to pass the parameters required for artifact deployment and service instantiation.
+
+Use the following Testkit builder methods:
+  - `withArtifactsDirectory(artifactsDirectory: Path)` to specify the directory in which the service
+  artifacts are located (usually, the `${project.build.directory}` = `target` directory)
+  - `withDeployedArtifact(artifactId: ServiceArtifactId, artifactFilename: String)` 
+  to specify an artifact to deploy in the test network
+  - `withService(artifactId: ServiceArtifactId, serviceName: String, serviceId: int)` 
+  to add a service instance of a previously deployed artifact with the given name and id.
+  If needed, initialization parameters may be supplied using an overloaded method.
+
+The parameters: artifactsDirectory, artifactId, artifactFilename may be passed to the test code
```

Comment:
```suggestion
Pass the `artifactsDirectory`, `artifactId`, `artifactFilename` parameters to the test code. Use the system properties
```

Reply:
"Pass ... to the test code" does not make much sense. You _somehow_ need to and will  pass them; but you could use the system properties for that.

Reply:
So it's more like "We recommend/Consider doing that using this particular mechanism *[because it is reliable enough and won't cause duplication of constants]* — but you could also do that differently"

Reply:
How about the following variant then?
> To pass the `artifactsDirectory`, `artifactId`, `artifactFilename` parameters to the test code, use the system properties from the build configuration properties, where the latter are already specified.